connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -180,6 +183,21 @@ export async function getRootNodesToSyncFromResources(
     nodesToSync
   );
 
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+
+  await concurrentExecutor(
```

Comment:
IMO resource is here to manage database entities, we should keep the logic of upserting into core outside of resource

Reply:
Understandable; I had understood resources could encapsulate business logic and do not necessarily directly map to models. and here we want to avoid people creating a folder-like resource without updating core
But agree there's a tradeoff so fine for me like this